platform/MCU/nRF52840/inc/platform_config.h

```diff
@@ -76,8 +74,7 @@
 #define PREPSTRING(x) PREPSTRING2(x)
 
 #if PLATFORM_ID == PLATFORM_XENON || PLATFORM_ID == PLATFORM_ARGON || PLATFORM_ID == PLATFORM_BORON || \
-    PLATFORM_ID == PLATFORM_XSOM || PLATFORM_ID == PLATFORM_ASOM || PLATFORM_ID == PLATFORM_BSOM || \
-    PLATFORM_ID == PLATFORM_B5SOM
+    PLATFORM_ID == PLATFORM_BSOM || PLATFORM_ID == PLATFORM_B5SOM
```

Comment:
How about including `hal_platform_nrf52840_config.h` and use `#if HAL_PLATFORM_NRF52840`? Or even we can move the `hal_platform_nrf52840_config.h` into this header file, as it is a platform-specific header file.

Reply:
This is located already in `nRF52840`-specific folder, so I'm going to leave as-is. For an unknown nRF52840-based platform this will cause a compiler error, which sounds like a good reminder to add a new platform here explicitly.